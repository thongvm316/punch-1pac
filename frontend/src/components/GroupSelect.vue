<template>
  <select
    class="form-select"
    @change="updateValue($event)"
  >
    <slot name="placeholder" />
    <option
      v-for="group in groups"
      :key="group.id"
      :value="group.id"
    >
      {{ group.name }}
    </option>
  </select>
</template>

<script>
import { mapState, mapActions } from 'vuex'

export default {
  name: 'GroupSelect',

  props: {
    value: {
      type: String,
      default: ''
    }
  },

  computed: {
    ...mapState('groups', ['groups'])
  },

  created() {
    this.getGroups()
  },

  methods: {
    ...mapActions('groups', ['getGroups']),

    updateValue(e) {
      this.$emit('input', e.target.value)
    }
  }
}
</script>
