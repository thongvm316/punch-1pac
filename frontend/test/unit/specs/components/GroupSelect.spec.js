import { shallowMount } from '@vue/test-utils'
import wrapperOps from '../../supports/wrapper'
import groupsData from '../../supports/fixtures/groups.api'
import GroupSelect from '@/components/GroupSelect'

const groups = [...groupsData.groups]
const scopedSlots = { placeholder: '<option>Choose a group</option>' }
const getGroups = jest.spyOn(GroupSelect.methods, 'getGroups')
const updateValue = jest.spyOn(GroupSelect.methods, 'updateValue')

const localWrapperOps = {
  ...wrapperOps,
  scopedSlots,
  computed: {
    groups: () => groups
  }
}

describe('GroupSelect.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(GroupSelect, localWrapperOps)
  })

  afterEach(() => { wrapper.vm.$destroy() })

  describe('when GroupSelect was mounted', () => {
    it('should render correctly', () => {
      expect(wrapper.exists()).toBeTruthy()
      expect(wrapper.isVueInstance()).toBeTruthy()
      expect(getGroups).toHaveBeenCalled()
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when updateValue', () => {
    it('should updateValue method called', () => {
      wrapper.find('select').trigger('change')

      expect(updateValue).toHaveBeenCalled()
      expect(wrapper.emitted('input')).toBeTruthy()
    })
  })
})
