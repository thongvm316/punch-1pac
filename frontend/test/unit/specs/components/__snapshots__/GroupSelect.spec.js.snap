// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupSelect.vue when GroupSelect was mounted should render correctly 1`] = `
<select class="form-select">
  <option>Choose a group</option>
  <option value="0">
    default
  </option>
  <option value="1">
    enum
  </option>
  <option value="2">
    dasda
  </option>
</select>
`;
